c-runtime/test/Systemtests/makefile

```diff
@@ -0,0 +1,152 @@
+RUNELST := /test/Systemtests/Conditions/TestElseStatement.out
+RUNELIFST := /test/Systemtests/Conditions/TestElifStatement.out
+RUNWHILEST := /test/Systemtests/Conditions/TestWhileStatement.out
+RUNCASTINT := /test/Systemtests/BuiltInFunctions/TestCastInt.out
+RUNCASTSTRING := /test/Systemtests/BuiltInFunctions/TestCastString.out
+RUNID := /test/Systemtests/BuiltInFunctions/TestId.out
+RUNPRINT := /test/Systemtests/BuiltInFunctions/TestPrint.out
+RUNTYPE := /test/Systemtests/BuiltInFunctions/TestType.out
+RUNFUNPARAMS := /test/Systemtests/FunctionCalls/TestFunctionParams.out
+RUNFUNPARAMSTHROWS := /test/Systemtests/FunctionCalls/TestFunctionParamsThrows.out
+RUNFUNBODY := /test/Systemtests/FunctionCalls/TestFunctionBody.out
+RUNFUNRET := /test/Systemtests/FunctionCalls/TestFunctionReturn.out
+RUNFUNREC := /test/Systemtests/FunctionCalls/TestFunctionRecursion.out
+RUNCLASSINIT := /test/Systemtests/Classes/TestClassInit.out
+RUNCLASSINITTHROWS := /test/Systemtests/Classes/TestClassInitThrows.out
+RUNCLASSMEMBERS := /test/Systemtests/Classes/TestClassMembers.out
+RUNCLASSINHERITANCE := /test/Systemtests/Classes/TestClassInheritance.out
+
+CXXFLAGS ?= -std=c++14
+
+.PHONY : all run echo test_declaration_and_assignment test_arithmetic_ops test_comparison_ops test_greater_equal_op test_greater_op test_less_equal_op test_less_op test_logical_ops test_if_statement test_else_statement test_elif_statement test_while_statement test_cast_int test_cast_string test_id test_print test_type test_function_params test_function_params_throws test_function_body test_function_return test_function_recursion test_class_init test_class_init_throws test_class_members test_class_inheritance clean
+
+all: test_declaration_and_assignment test_arithmetic_ops test_comparison_ops test_greater_equal_op test_greater_op test_less_equal_op test_less_op test_logical_ops test_if_statement test_else_statement test_elif_statement test_while_statement test_cast_int test_cast_string test_print test_type test_function_params test_function_params_throws test_function_body test_function_return test_function_recursion test_class_init test_class_init_throws test_class_members test_class_inheritance run
+
+run:
+	.$(RUNDECLA)
+	.$(RUNARITOPS)
+	.$(RUNCOMPOPS)
+	.$(RUNGEOP)
+	.$(RUNGTOP)
+	.$(RUNLEOP)
+	.$(RUNLTOP)
+	.$(RUNLOGOPS)
+	.$(RUNIFST)
+	.$(RUNELST)
+	.$(RUNELIFST)
+	.$(RUNWHILEST)
+	.$(RUNCASTINT)
+	.$(RUNCASTSTRING)
+	.$(RUNPRINT)
+	.$(RUNTYPE)
+	.$(RUNFUNPARAMS)
+	.$(RUNFUNPARAMSTHROWS)
+	.$(RUNFUNBODY)
+	.$(RUNFUNRET)
+	.$(RUNFUNREC)
+	.$(RUNCLASSINIT)
+	.$(RUNCLASSINITTHROWS)
+	.$(RUNCLASSMEMBERS)
+	.$(RUNCLASSINHERITANCE)
+
+echo:
+	@echo ${C_FILES}
+	@echo ${TESTDIR}
+	@echo ${RUN}
+
+test_declaration_and_assignment:
+	g++-13 -I $(INCLUDEDIR) -I $(INCLUDEDIRTEST) -o $(TESTDIR)/Systemtests/DeclarationAndAssignment/TestDeclarationAndAssignment.out -w -fpermissive $(TESTDIR)/Systemtests/DeclarationAndAssignment/TestDeclarationAndAssignment.c* $(C_FILES_TEST)
```

Comment:
Hmmm. Den Standard 13 gibt es nicht. Ich hab es mal mit "-std=c++17" probiert, und ein "make test" im c-runtime-Ordner gibt mir tausende Fehler der Art

```
./src/builtins-setup.c:265:5: error: no matching function for call to '__mpy_hash_map_put'
    __mpy_hash_map_put(typeObjectAttrs, "__str__", __MPyFunc_Object_str);
```

Reply:
Ja die Fehler sind mir bekannt, leider konnte ich den Ursprung während der Bearbeitung nicht finden. Hier ist, was ich darüber weiß.

Ursprünglich habe ich mich für g++-13 entschieden, weil es die neuste Version vom damaligen GitHub macOS image war. Dabei gibt die 13 glaube ich nicht den Standard der gesetzt wird sonder die Version des Build-Tools? (Korrigier mich hier gerne) Mir ist erst aufgefallen, dass es Probleme gibt, als ich das Ubuntu Image mit in den workflow eingebunden habe. Du hast mich bei einem Meeting auf die unterschiedlichen Versionen von den Runnern aufmerksam gemacht.

* macOS : GCC 13 (Homebrew GCC 13.3.0) - available by gcc-13 alias 
* Ubuntu 22.04 : gcc 4:11.2.0-1ubuntu1
* Ubuntu 24.04 (beta) :  gcc 4:13.2.0-7ubuntu1

Eigentlich sollte es damit keine Probleme geben? Aber ich habe es bis jetzt immer auf die älteren Versionen geschoben, weil ich keine bessere Erklärung gefunden habe. Mit der 24.04 beta kommt jetzt gcc 13 vielleicht wird es damit ja auch unter Ubuntu gehen... Wobei ich nicht weiß ob das Alias g++-13 da funktioniert.

Alles in einem könntest du mal schauen, welche Version von gcc bei dir installiert ist? Wahrscheinlich ist es nicht die  4:13.x.x oder ?

Reply:
Das Schräge ist, dass ich tatsächlich via `-std=` den c++13-Standard eingestellt hatte (direkt im Makefile) - und das lief vor die Wand.

Muss ich wohl mal in Ruhe nachschauen, was da wohl ist.

Reply:
Lief es denn durch, wenn du den Standard auf `-std=c++14` setzt und ein g++ mit Version 13. nutzt? Bei mir funktioniert es mit `g++-13 (Homebrew GCC 13.2.0) 13.2.0`  Ich kann in den folgenden Tagen mal testen, ob es unter Ubuntu 24 funktioniert. Am besten wären die Github Actions. Diese scheinen aber nicht zu starten, wie ich grade sehe...

`The job was not started because recent account payments have failed or your spending limit needs to be increased. Please check the 'Bil...`